ReactAndroid/src/main/java/com/facebook/react/packagerconnection/PackagerConnectionSettings.java

```diff
@@ -54,10 +54,21 @@ public String getDebugServerHost() {
   }
 
   public void setDebugServerHost(String host) {
+    mDebugServerHost = host;
     mPreferences.edit().putString(PREFS_DEBUG_SERVER_HOST_KEY, host).apply();
   }
 
+  public void setDebugServerHostWithoutPersisting(final String host) {
+    mDebugServerHost = host;
+  }
+
   public String getInspectorServerHost() {
+    // Check host setting first. If empty try to detect emulator type and use default
+    // hostname for those
```

Comment:
The comment seems to be copy-pasted from `getDebugServerHost()` and is misleading. We are not trying to detect the emulator type here.